src/mon/MonClient.cc

```diff
@@ -397,13 +369,13 @@ int MonClient::init()
     return r;
   }
 
-  rotating_secrets = new RotatingKeyRing(cct, cct->get_module_type(), keyring);
+  rotating_secrets.reset(new RotatingKeyRing(
+			   cct, cct->get_module_type(), keyring.get()));
 
   initialized = true;
 
-  timer.init();
   finisher.start();
-  schedule_tick();
+  timer.add_event(tick_time(), &MonClient::timer, this);
```

Comment:
SafeTimer doesn't spawn a thread until `init()`, so this timer should probably start suspended as well.